### ISSUE TITLE
Add 'x-default' to site group that does not contain a primary site?

### ISSUE THREAD

**Issue by tibbis** [nystudio107/craft-seomatic#1555]
### Question
The x-default is added perfectly to a site group where the primary site is, however if site groups logically separates them as sister-sites, the x-default hreflang tag is not added. 

As craft does not have a feature to add primary site per site group, I added a global field my self to each site group to tell if its a "default site or not". 

Now I want to instruct Seomatic this and override "x-default" to the specific site if it doesn't exist already, what would be the easiest way to do this? I imagine it will be difficult to this to the sitemaps however?

Thanks

### Additional context
Craft cms 5 pro

**Comment by khalwat** [comment]
Did you follow the site setup/config mentioned here?

https://nystudio107.com/docs/seomatic/multi-site.html

**Comment by tibbis** [comment]
Yes I have, I understand the sister site structure. But there is no way of having different default/primary site in each site group.

**Comment by khalwat** [comment]
Yeah the problem is that SEOmatic has no way of determining what the default site should actually be in this case... because Craft itself has no such mechanism. Which of course you noticed yourself, because you had to add a field to designate it.

So it sounds like you're doing a relatively custom setup, so you'll need to write some custom code. Fortunately, SEOmatic has a robust Twig & PHP API that you can use to do this:

https://nystudio107.com/docs/seomatic/using/link-meta.html#link-meta-object-examples

Here's a specific example for you:

```twig
    {% set alt = seomatic.link.get('alternate') %}
    {% do alt.href([
            'http://example.com',
            'http://example.com/es'
        ]).hreflang([
            'x-default',
            'es',
        ])
    %}
```

see also: https://github.com/nystudio107/craft-seomatic/issues/370#issuecomment-484095678

**Comment by khalwat** [comment]
While this is arguably a limitation of SEOmatic, Craft's "Sites" system doesn't provide any real context that SEOmatic can infer from, because it doesn't provide any structure we can key off of, other than one "primary" site.

The alternative would be a relatively complicated UX that SEOmatic would have to provide to give said context. I feel that this is more in the purview of the CMS itself to define sites and their relationship between eachother.
